infrastructure/lambda-get-document-nrl.tf

```diff
@@ -0,0 +1,39 @@
+resource "aws_api_gateway_resource" "document_ref_get" {
+  rest_api_id = aws_api_gateway_rest_api.ndr_doc_store_api.id
+  parent_id   = module.create-doc-ref-gateway.gateway_resource_id
+  path_part   = "{id}"
+}
+
+resource "aws_api_gateway_method" "document_ref_get" {
+  rest_api_id      = aws_api_gateway_rest_api.ndr_doc_store_api.id
+  resource_id      = aws_api_gateway_resource.document_ref_get.id
+  http_method      = "GET"
+  authorization    = "NONE"
+  api_key_required = true
+  request_parameters = {
+    "method.request.path.id" = true
+  }
+}
+
+module "get-doc-nrl-lambda" {
+  source  = "./modules/lambda"
+  name    = "GetDocumentReference"
+  handler = "handlers.get_doc.lambda_handler"
+  iam_role_policies = [
```

Comment:
This block will need refactoring when [PRMP-1221](https://nhse-dsic.atlassian.net/browse/PRMP-1221) is merged.